src/Fixer/PhpUnit/PhpUnitTestAnnotationFixer.php

```diff
@@ -291,26 +291,36 @@ private function getDockBlockIndex(Tokens $tokens, $index)
         return $index;
     }
 
+    /**
+     * @param string $functionName
+     *
+     * @return bool
+     */
+    private function hasTestPrefix($functionName)
+    {
+        if (!$this->startsWith('test', $functionName) || strlen($functionName) < 5) {
+            return false;
+        }
+
+        $nextCharacter = $functionName[4];
+
+        return $nextCharacter === strtoupper($nextCharacter);
+    }
+
     /**
      * @param string $functionName
      *
      * @return string
      */
     private function removeTestFromFunctionName($functionName)
     {
-        if ($this->startsWith('test_', $functionName)) {
-            if (is_numeric(substr($functionName, 6, 1))) {
-                return $functionName;
-            }
+        $remainder = preg_replace('/^test_?/', '', $functionName);
 
-            return  substr($functionName, 5);
-        }
-        if (is_numeric(substr($functionName, 5, 1))) {
+        if ('' === $remainder || is_numeric($remainder[0])) {
```

Comment:
Can't remove the prefix if the remainder is an empty string, or the first character is a digit.